sync.js

```diff
@@ -24,6 +24,8 @@ var sync = module.exports = function(){
   }
 }
 
+var nextTick = setImmediate || process.nextTick || function(cb) { setTimeout(cb, 0) }
```

Comment:
This should be `((typeof setImmediate !== 'undefined') && setImmediate) || process.nextTick…`, otherwise we'll get a reference error if `setImmediate` isn't defined.